nexus/db-queries/src/db/datastore/silo_group.rs

```diff
@@ -41,7 +41,12 @@ impl DataStore {
         Ok(diesel::insert_into(dsl::silo_group)
             .values(silo_group)
             .on_conflict((dsl::silo_id, dsl::external_id))
-            .filter_target(dsl::time_deleted.is_null())
+            // This used to have a `filter_target` condition on
+            // `dsl::time_deleted.is_null()`, but that's been removed since
+            // filter_target doesn't work with CockroachDB:
+            // https://github.com/oxidecomputer/omicron/issues/5047.
+            //
+            // XXX: is this correct?
```

Comment:
@jmpesp I was trying to figure out was this was trying to do -- how much do we care about deleted rows here?

Reply:
I'm also wondering whether returning the silo is the right thing to do -- with `ON CONFLICT DO NOTHING`, no rows are returned if the row already exists, and any code that attempts to get an item will fail.

Reply:
This filter_target allows us to delete a silo group, then re-create another one later (through JIT user and group creation) using the same group string. I don't think it's correct to remove this check, but it also looks like the original `filter_target` was being ignored :(

Reply:
> I'm also wondering whether returning the silo is the right thing to do -- with `ON CONFLICT DO NOTHING`, no rows are returned if the row already exists, and any code that attempts to get an item will fail.

Yeah, this is probably not the right thing to do then.

Reply:
Digging into this more -- it turns out that this is actually required if the constraint you're matching against has a WHERE clause. In this case, the WHERE clause is https://github.com/oxidecomputer/omicron/blob/6c9c9b73d82c91389f0c8ccb46e33aaaf0afdf03/schema/crdb/dbinit.sql#L706-L707.

Also, you can truly pass in _anything_ which implies the WHERE clause in the constraint, according to Aristotelian (classical) logic. For example, you can pass in `WHERE false = true` or `WHERE 1 = 2`, since in Aristotelian logic [a contradiction implies any proposition](https://en.wikipedia.org/wiki/Principle_of_explosion).

Reply:
Replaced with `as_partial_index`.